tiledb/sm/query/ast/CMakeLists.txt

```diff
@@ -0,0 +1,131 @@
+#
+# tiledb/sm/query/ast/CMakeLists.txt
+#
+#
+# The MIT License
+#
+# Copyright (c) 2021-2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+include(common NO_POLICY_SCOPE)
+
+################################################################
+#
+# This is temporary until test_support is widely deployed
+#
+################################################################
+
+
+set(TILEDB_TEST_SUPPORT_SOURCES
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../../test/src/helpers.h
```

Comment:
Unit tests shouldn't rely on files in the test directly like this. Can we either switch to using the object library for the helpers or reduce the scope of whats needed for the unit tests?

Reply:
I think we tried this, but failed to do so. @lums658 care to elaborate?

Reply:
The problem is that cmake pulls in the individual unit tests before it sets up the helpers object library.  So we have to pull in the helpers ourselves.  This would be a problem for any individual unit test that wants to use helpers.  We probably shouldn't be having this dependency, but there is shared code between some things in test/src and some things in query/ast/test.  We could probably factor the act-specific stuff out of helpers and just put it in, say, ast_helpers -- but I think there are other dependencies.  Best solution might be to move helper library up in the cmake chain.

Reply:
moving helpers object library build position is in process (https://github.com/TileDB-Inc/TileDB/pull/3187)